python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -314,50 +327,149 @@ cdef class Column:
             c_result = make_column_from_scalar(dereference(slr.get()), size)
         return Column.from_libcudf(move(c_result))
 
-    @staticmethod
-    def from_cuda_array_interface_obj(object obj):
-        """Create a Column from an object with a CUDA array interface.
+    @classmethod
+    def from_array_interface(cls, obj):
+        """
+        Create a Column from an object implementing the NumPy Array Interface.
+
+        Parameters
+        ----------
+        obj : object
+            Must implement the `__array_interface__` protocol.
+
+        Raises
+        ------
+        NotImplementedError
+            This method is not yet implemented.
+        """
+        raise NotImplementedError(
+            "Converting to a pylibcudf Column is not yet implemented."
+        )
+
+    @classmethod
+    def from_cuda_array_interface(cls, obj):
+        """
+        Create a Column from an object implementing the CUDA Array Interface.
 
         Parameters
         ----------
         obj : object
-            The object with the CUDA array interface to create a column from.
+            Must implement the ``__cuda_array_interface__`` protocol.
 
         Returns
         -------
         Column
             A Column containing the data from the CUDA array interface.
 
-        Notes
-        -----
-        Data is not copied when creating the column. The caller is
-        responsible for ensuring the data is not mutated unexpectedly while the
-        column is in use.
+        Raises
+        ------
+        ValueError
+            If the object is not 1D or 2D, or is not C-contiguous.
+        ImportError
+            If CuPy is not installed.
         """
-        data = gpumemoryview(obj)
-        iface = data.__cuda_array_interface__
-        if iface.get('mask') is not None:
-            raise ValueError("mask not yet supported.")
+        iface = getattr(obj, "__cuda_array_interface__", None)
+        if iface is None:
+            raise TypeError("Object does not implement __cuda_array_interface__")
```

Comment:
This is a classic case where we should prefer a try-except. It'll be faster in the majority of relevant cases where someone passes an object that actually has a CAI. If a user passes something that does not, we're going to error any way, so we can afford the overhead of exception handling. 
```suggestion
        try:
            iface = obj.__cuda_array_interface__
        except AttributeError:
            raise TypeError("Object does not implement __cuda_array_interface__")
```